astroquery/mast/core.py

```diff
@@ -1394,5 +1416,417 @@ def download_products(self, products, download_dir=None,
+                  'radius': radius.deg}
+
+        # Hsc specific parameters (can be overridden by user)
+        params['nr'] = 50000
+        params['ni'] = 1
+        params['magtype'] = 1
+
+        # galex specific parameters (can be overridden by user)
+        params['maxrecords'] = 50000
+
+        # adding additional parameters
+        for prop, value in kwargs.items():
+            params[prop] = value
+
+        return self.service_request_async(service, params, pagesize, page)
+
+    @class_or_instance
+    def query_object_async(self, objectname, radius=0.2*u.deg, catalog="Hsc",
+                           pagesize=None, page=None, **kwargs):
+        """
+        Given an object name, returns a list of catalog entries.
+        See column documentation for specific catalogs `here <https://mast.stsci.edu/api/v0/pages.html>`__.
+
+        Parameters
+        ----------
+        objectname : str
+            The name of the target around which to search.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.2 degrees.
+            The string must be parsable by `astropy.coordinates.Angle`.
+            The appropriate `~astropy.units.Quantity` object from
+            `astropy.units` may also be used. Defaults to 0.2 deg.
+        catalog : str, optional
+            Default HSC.
+            The catalog to be queried.
+        pagesize : int, optional
+            Default None.
+            Can be used to override the default pagesize for (set in configs) this query only.
+            E.g. when using a slow internet connection.
+        page : int, optional
+            Defaulte None.
+            Can be used to override the default behavior of all results being returned
+            to obtain a specific page of results.
+        **kwargs
+            Catalog-specific keyword args.
+            These can be found in the `service documentation <https://mast.stsci.edu/api/v0/_services.html>`__.
+            for specific catalogs. For example one can specify the magtype for an HSC search.
+
+        Returns
+        -------
+        response: list of ``requests.Response``
+        """
+
+        coordinates = self._resolve_object(objectname)
+
+        return self.query_region_async(coordinates, radius, catalog, pagesize, page, **kwargs)
+
+    @class_or_instance
+    def query_criteria_async(self, catalog, pagesize=None, page=None, **criteria):
+        """
+        Given an set of filters, returns a list of catalog entries.
+        See column documentation for specific catalogs `here <https://mast.stsci.edu/api/v0/pages.htmll>`__.
+
+        Parameters
+        ----------
+        pagesize : int, optional
+            Can be used to override the default pagesize.
+            E.g. when using a slow internet connection.
+        page : int, optional
+            Can be used to override the default behavior of all results being returned to obtain
+            one sepcific page of results.
```

Comment:
typo

Reply:
I fixed this, I don't know why it isn't shown as outdated now.